components/dashboard/src/settings/EnvironmentVariables.tsx

```diff
@@ -4,12 +4,195 @@
  * See License-AGPL.txt in the project root for license information.
  */
 
+import { UserEnvVarValue } from "@gitpod/gitpod-protocol";
+import { useEffect, useRef, useState } from "react";
+import ContextMenu from "../components/ContextMenu";
+import Modal from "../components/Modal";
+import { getGitpodService } from "../service/service";
 import { SettingsPage } from "./SettingsPage";
+import ThreeDots from '../icons/ThreeDots.svg';
+
+interface EnvVarModalProps {
+    envVar: UserEnvVarValue;
+    onClose: () => void;
+    save: (v: UserEnvVarValue) => void;
+    validate: (v: UserEnvVarValue) => string;
+}
+
+function AddEnvVarModal(p: EnvVarModalProps) {
+    const [ev, setEv] = useState({...p.envVar});
+    const [error, setError] = useState('');
+    const ref = useRef(ev);
+
+    const update = (pev: Partial<UserEnvVarValue>) => {
+        const newEnv = { ...ref.current, ... pev};
+        setEv(newEnv);
+        ref.current = newEnv;
+    };
+
+    useEffect(() => {
+        setEv({...p.envVar});
+        setError('');
+    }, [p.envVar]);
+
+    const isNew = !p.envVar.id;
+    let save = () => {
+        const v = ref.current;
+        const errorMsg = p.validate(v);
+        if (errorMsg !== '') {
+            setError(errorMsg);
+            return false;
+        } else {
+            p.save(v);
+            p.onClose();
+            return true;
+        }
+    };
+
+    return <Modal visible={true} onClose={p.onClose} onEnter={save}>
+        <h3 className="pb-2">{isNew ? 'New' : 'Edit'} Variable</h3>
+        <div className="border-t -mx-6 px-6 py-2 flex flex-col">
+            {error ? <div className="bg-gitpod-kumquat-light rounded-md p-3 text-red-500 text-sm">
+                {error}
+            </div> : null}
+            <div className="mt-4">
+                <h4>Name</h4>
+                <input className="w-full" type="text" value={ev.name} onChange={(v) => { update({name: v.target.value}) }} />
+            </div>
+            <div className="mt-4">
+                <h4>Value</h4>
+                <input className="w-full" type="text" value={ev.value} onChange={(v) => { update({value: v.target.value}) }} />
+            </div>
+            <div className="mt-4">
+                <h4>Scope</h4>
+                <input className="w-full" type="text" value={ev.repositoryPattern} placeholder="org/project" 
+                    onChange={(v) => { update({repositoryPattern: v.target.value}) }} />
+            </div>
+            <div className="mt-3">
+                <p>You can pass a variable for a specific org/project or use wildcard characters (*/*) to make it available in more projects.</p>
```

Comment:
**suggestion**: I think we could skip the org/project reference here as the placehodler in the input already mentions this. Your call. 📞 

```suggestion
                <p>You can pass a variable for a specific project or use wildcard character to make it available in more projects or groups.</p>
```

Reply:
I don't think wildcard character is easily understandable. So I'd like to give an example on that. WDYT?

Reply:
DEAL